contributors/design-proposals/raw-block-pv.md

```diff
@@ -0,0 +1,568 @@
+  PV = block, PVC = file: destroy the filesystem
+  PV = file, PVC = file: delete the files
+
+  The last design point is block devices should be able to be fully restricted by the admin in accordance with how inline volumes 
+  are today. Ideally, the admin would want to be able to restrict either raw-local devices and or raw-network attached devices.
+  
+# Proposed API Changes
+   
+## Persistent Volume Claim API Changes:
+In the simplest case of static provisioning, a user asks for a volumeType of block. The binder will only bind to a PV defined 
+with the same volumeType.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  volumeType: block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+For dynamic provisioning and the use of the storageClass, the user also specifically defines the intent of the volume by 
+indicating the volumeType as block. The provisioner for this class will validate whether or not it supports block and return
+an error if it does not.
+
+```
+kind: PersistentVolumeClaim
+apiVersion: v1
+metadata:
+name: myclaim
+spec:
+  storageClassName: local-fast 
+  volumeType: block #proposed API change
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 80Gi 
+```
+
+## Persistent Volume API Changes:
+For static provisioning the admin creates the volume and also is intentional about how the volume should be consumed. For backwards
+compatibility, the absence of volumeType will default to file which is how volumes work today, which are formatted with a filesystem depending on the plug-in chosen. Recycling will not be a supported reclaim policy. Once the user deletes the claim against a PV, the volume will be scrubbed according to how it was bound. The path value in the local PV definition would be overloaded to define the path of the raw block device rather than the fileystem path.
+```
+kind: PersistentVolume
+apiVersion: v1
+metadata:
+name: local-raw-pv
+spec:
+  volumeType: block #proposed API change
+  capacity:
+    storage: 100Gi
+  local:
+    path: /dev/xvdc 
+  accessModes:
+    - ReadWriteOnce
+  persistentVolumeReclaimPolicy: Delete 
+```
+
+## Storage Class API Changes:
+For dynamic provisioning, it is assumed that values pass in the parameter section are opaque, thus the introduction of utilizing
+fsType in the StorageClass can be used by the provisioner to indicate how to create the volume. The proposal for this value is
+defined here:
+https://github.com/kubernetes/kubernetes/pull/45345 
+Therefore, a provisioner could potentially provision a block device and install the filesystem onto it by indicating the volumeType
+as 'block' but the fsType as 'xfs'.
+
+```
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: block-volume
+provisioner: kubernetes.io/local-block-glusterfs
+parameters:
+  volumeType: block #opaque value  / plug-in dependent
+  fsType: xfs
```

Comment:
@lpabon Because of the technical capability of a plugin and provisioner that can do both and triggers off that value as a flag to do one or the other. This paper calls it out as a line to indicate that value is not going to validate and fail if it doesn't match